### PR TITLE
ClinicaDL cleaning 

### PR DESCRIPTION
**Pull request by camillebrianceau**

### CODE REVIEW

File: clinicadl/train/train_utils.py [hunk 1]
```diff
@@ -76,7 +77,7 @@ def build_train_dict(config_file: Path, task: str) -> Dict[str, Any]:
 
     elif config_file.suffix == ".json":
         train_dict = read_json(config_file)
-        train_dict = change_str_to_path(train_dict)
+        # train_dict = change_str_to_path(train_dict)
```
Comment (ncassereau) [inline clinicadl/train/train_utils.py]: is this on purpose that it is not replaced with `path_decoder` ?

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 2]
```diff
@@ -2077,7 +2075,7 @@ def _check_args(self, parameters):
                     f"No value was given for {arg}."
                 )
         self.parameters = add_default_values(parameters)
-        self.parameters = change_str_to_path(parameters)
+        # self.parameters = change_str_to_path(parameters)
```
Comment (ncassereau) [inline clinicadl/utils/maps_manager/maps_manager.py]: unless i'm missing something, I believe that the path_decoder is never called. Is this intended ?

File: clinicadl/utils/maps_manager/maps_manager_utils.py [hunk 3]
```diff
@@ -211,54 +214,8 @@ def change_str_to_path(
                 or key.endswith("json")
                 or key.endswith("location")
             ):
-                if value == "":
+                if value == "" or value is False:
```
Comment (ncassereau) [inline clinicadl/utils/maps_manager/maps_manager_utils.py]: Does this function still have a reason to exist now that `path_decoder` is a thing ?

**Review by ncassereau — COMMENTED** [review]
A few lines of commented-out code still remain. Other than that, looks good to me.

File: clinicadl/generate/generate.py [hunk 4]
```diff
@@ -383,18 +378,32 @@ def generate_shepplogan_dataset(
...
-        extract_json: name of the JSON file in which generation details are stored.
-        samples: number of samples generated per class.
-        smoothing: if True, an additional random smoothing is performed on top of all operations on each image.
+    Parameters
+    ----------
+    output_dir: Path
+        Path to the CAPS created.
+    img_size: int
+        Size of the square image.
+    n_procs: int
```
Comment (thibaultdvx) [inline clinicadl/generate/generate.py]:
```suggestion
    n_proc: int
```

File: clinicadl/generate/generate.py [hunk 5]
```diff
@@ -71,6 +70,8 @@ def generate_random_dataset(
         Folder containing the synthetic dataset in CAPS format.
     n_subjects: int
         Number of subjects in each class of the synthetic dataset
+    n_procs: int
```
Comment (thibaultdvx) [inline clinicadl/generate/generate.py]:
```suggestion
    n_proc: int
```

File: clinicadl/generate/generate.py [hunk 6]
```diff
@@ -211,6 +205,8 @@ def generate_trivial_dataset(
         Folder containing the synthetic dataset in CAPS format.
     n_subjects: int
         Number of subjects in each class of the synthetic dataset.
+    n_procs: int
```
Comment (thibaultdvx) [inline clinicadl/generate/generate.py]:
```suggestion
    n_proc: int
```

File: clinicadl/generate/generate.py [hunk 7]
```diff
@@ -696,40 +709,43 @@ def generate_artifacts_dataset(
...
+        Translation range in mm of simulated movements.
+    rotation : List
+        Rotation range in degree of simulated movement.
+    num_transformes: int
+        Number of simulated movements.
+    gamma: List
+        Gamma range of simulated contrast.
+    noise_std: List
+        Stadndard deviation of simulated noise.
+
```
Comment (thibaultdvx) [inline clinicadl/generate/generate.py:748]: `Contrast`, `gamma` and `motion` parameters are missing.

File: clinicadl/generate/generate_utils.py [hunk 8]
```diff
@@ -318,7 +318,7 @@ def generate_shepplogan_phantom(
     return img
 
 
-def mask_processing(mask, percentage, sigma):
+def mask_processing(mask, percentage: float, sigma: float):
```
Comment (thibaultdvx) [inline clinicadl/generate/generate_utils.py:321]: `mask`'s type?

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 9]
```diff
@@ -640,7 +637,7 @@ def interpret(
     ###################################
     # High-level functions templates  #
     ###################################
-    def _train_single(self, split_list=None, resume=False):
+    def _train_single(self, split_list: List[int] = None, resume: bool = False):
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
    def _train_single(self, split_list: Optional[List[int]] = None, resume: bool = False):
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 10]
```diff
@@ -2827,7 +2825,7 @@ def _init_split_manager_ssda(self, caps_dir, tsv_dir, split_list=None):
 
         return split_class(**kwargs)
 
-    def _init_task_manager(self, df=None, n_classes=None):
+    def _init_task_manager(self, df: pd.DataFrame = None, n_classes: int = None):
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
    def _init_task_manager(self, df: Optional[pd.DataFrame] = None, n_classes: Optional[int] = None):
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 11]
```diff
@@ -2931,60 +2929,72 @@ def get_group_info(
...
+                )
+        return self.get_state_dict(split=split, selection_metric=selection_metric)[
+            "epoch"
+        ]
 
     def get_state_dict(
-        self, split=0, selection_metric=None, network=None, map_location=None
+        self,
+        split=0,
+        selection_metric: str = None,
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
        selection_metric: Optional[str] = None,
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 12]
```diff
@@ -2931,60 +2929,72 @@ def get_group_info(
...
+        network: int = None,
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
        network: Optional[int] = None,
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 13]
```diff
@@ -2931,60 +2929,72 @@ def get_group_info(
...
+        map_location: str = None,
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
        map_location: Optional[str] = None,
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 14]
```diff
@@ -3016,7 +3026,12 @@ def get_state_dict(
         return torch.load(model_path, map_location=map_location)
 
     def get_prediction(
-        self, data_group, split=0, selection_metric=None, mode="image", verbose=False
+        self,
+        data_group: str,
+        split: int = 0,
+        selection_metric: str = None,
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
        selection_metric: Optional[str] = None,
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 15]
```diff
@@ -3053,7 +3068,12 @@ def get_prediction(
         return df
 
     def get_metrics(
-        self, data_group, split=0, selection_metric=None, mode="image", verbose=True
+        self,
+        data_group: str,
+        split: int = 0,
+        selection_metric: str = None,
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
        selection_metric: Optional[str] = None,
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 16]
```diff
@@ -3087,14 +3107,14 @@ def get_metrics(
...
-        split=0,
-        selection_metric=None,
-        verbose=True,
-        participant_id=None,
-        session_id=None,
-        mode_id=0,
+        data_group: str,
+        name: str,
+        split: int = 0,
+        selection_metric: str = None,
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
        selection_metric: Optional[str] = None,
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 17]
```diff
@@ -3087,14 +3107,14 @@ def get_metrics(
...
+        verbose: bool = True,
+        participant_id: str = None,
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
        participant_id: Optional[str] = None,
```

File: clinicadl/utils/maps_manager/maps_manager.py [hunk 18]
```diff
@@ -3087,14 +3107,14 @@ def get_metrics(
...
+        session_id: str = None,
```
Comment (thibaultdvx) [inline clinicadl/utils/maps_manager/maps_manager.py]:
```suggestion
        session_id: Optional[str] = None,
```

**Review by thibaultdvx — COMMENTED** [review]
Some minor suggestions

**Review by thibaultdvx — COMMENTED** [review]
LGTM

**Review by thibaultdvx — COMMENTED** [review]
LGTM
